README.md

```diff
@@ -1,2 +1,276 @@
-# example-application
-Example out-of-tree application that is also a module
+![Zephyr Project logo](https://github.com/zephyrproject-rtos/zephyr/blob/v1.14-branch/doc/images/Zephyr-Project.png)
+
+
+# Repository Overview
+
+The Zephyr Project is a scalable real-time operating system (RTOS) supporting
+multiple hardware architectures, optimized for resource constrained devices,
+and built with security in mind.
+
+The Zephyr build system supports integration of externally maintained
+projects called *modules*. This integration is configured through the
+settings in a *module.yml* file in a :file:`zephyr/` folder at the root
+of the repository.  See the [Zephyr Modules (External projects)][5]
+for a high-level overview of the build system module mechanism.
+
+
+## Scope
+
+The Zephyr eXample Application (ZXA) module is supplementary to the Zephyr
+repository.  This provides:
+
+1. A reference implementation of using the Zephyr module mechanism.
+
+1. A repository for holding example code and config files outside of the
+   Zephyr repository, including:
+
+   - `west` manifest (in the default location)
+   - projects/samples/tests
+   - applications/subsystems/drivers/interface declarations
+   - devicetree source files
+   - boards
+   - scripts
+
+1. A test case for CI to monitor Zephyr build system support for modules.
+
+
+   _WARNING:_ The Zephyr repository is intentionally
+   **independent of and unaware of this repository**.
+
+
+# Getting Started
+
+## Creating a New Module
+
+This module is designed to be a boilerplate for other modules simply by:
+
+1. cloning
+
+1. stripping out any unnecessary content (e.g. no drivers or subsystems)
+
+  - If you don't have any non-sample, non-test projects:
+
+    1. remove the `zephyr/projects/` subdirectory
```

Comment:
As discussed in a meeting recently, the `zephyr/` folder should only contain `module.yml`. Everything else should be in the root of the repo, because this is an example for a manifest repo in star topology, not an example on how to make an existing repository into a zephyr module.

Reply:
@carlescufi That discussion must not have been one I was present for. What is the reasoning behind it?

Declaring that this is **not** an example on how to make an existing repository into a zephyr module is actually a change in scope.  Nothing in our discussions or the issue description has precluded my design.

Please recall that I initiated this effort with an email, and one of my use cases was to extend existing repositories with minimal impact (whether proprietary or existing open source, specifically [project-chip](https://github.com/project-chip/connectedhomeip) and [bacnet-stack](https://github.com/bacnet-stack/bacnet-stack)). My primary goal was and is to ensure that a minimal impact solution for my use case is perpetually supported by the Zephyr build system.  After much consideration of the options I ended up being left with this configuration.  

When you provide me a different repository for:
- an example of extending an existing repository into a Zephyr module
- an example repository for non-trivial examples, e.g.:
    - namespace management
    - completely set up a user-space thread for C++ code
    - CI and code coverage on the module
    - local & workspace-wide document generation (where the workspace documentation/logo/theme is in the app module)
    - app-module requirements & artifact (traceability matrix) generation
then I'll move these folders up in this PR.

Reply:
The scope, as agreed with @tejlmand and @mbolivar-nordic was clearly described in the body of this GitHub issue:
https://github.com/zephyrproject-rtos/zephyr/issues/28438

If you think that the original scope does not cover the entirety of your use cases that is fine, I suggest you create your own set of complex repositories that do that in your own GitHub account, which you can feel free to share with others. But when it comes to repositories contained in the zephyrproject-rtos GitHub organization I will insist on starting small and then considering further development on the foundations that have been approved by several members of the community. Indeed, if you manage to rally support for these complex scenarios later down the line I will have no problem in including them upstream.